File: java/Search Insert Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int c=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]==target)
            { c++;
                return i;
                
            }
        }
            if(c==0)
            {
                for(int i=0;i<nums.length;i++)
                    if(target<nums[i])
                        return i;
                
            } return nums.length;
        }
        
    }

File: c++/SearchInsertPosition.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        int s =0;
        int e = nums.size() -1;
    
        while(s<e){
            
            int mid  =  s + (e-s)/2 ;
           
            if(nums[mid] == target){
                return mid;
                
            }else if(nums[mid] < target){
                s = mid + 1;
                
            }else{
                e = mid ;
            }
        }
    return nums[s] < target ? s + 1: s;
     
    }
};